pkg/kubelet/nodestatus/setters.go

```diff
@@ -194,8 +194,8 @@ func NodeAddress(nodeIP net.IP, // typically Kubelet.nodeIP
 			// 3) Lookup the IP from node name by DNS
 			// 4) Try to get the IP from the network interface used as default gateway
 			//
-			// For steps 3 and 4, IPv4 addresses are preferred to IPv6 addresses
-			// unless nodeIP is "::", in which case it is reversed.
+			// For steps 3 and 4, IPv4 addresses are used unless
+			// nodeIP is "::", in which case IPv6 addresses are used.
```

Comment:
This is a backward-compatibility break. In particular, it means that in an IPv6-only environment, you now *must* pass `--node-ip ::` (or `--node-ip` with a specific IPv6 address).

Reply:
slightly off topic but I'm reverting kind leveraging auto detection until there's a plan here.

Reply:
no, kubelet is not aware of its IP family

Reply:
Fail enough, you'll fix this with your kep ;)